internal/biome/docker.go

```diff
@@ -253,6 +253,35 @@ func (c *Container) Run(ctx context.Context, invoke *Invocation) error {
 		return fmt.Errorf("run in container %s: %w", c.id, err)
 	}
 
+	// Unfortunately, there's no way to signal a particular exec:
+	// https://github.com/moby/moby/issues/9098
+	//
+	// So if the Context is cancelled, we restart the container, which may take
+	// down other running processes, but usually the biome is going down anyway.
+	runDone := make(chan struct{})
+	interrupterDone := make(chan struct{})
+	go func() {
+		defer close(interrupterDone)
+		select {
+		case <-ctx.Done():
+			log.Infof(ctx, "Interrupted. Stopping container %s...", c.id)
+			if err := c.client.StopContainer(c.id, 10); err != nil {
+				log.Warnf(ctx, "Could not stop container %s: %v", c.id, err)
+				return
+			}
+			err := c.client.StartContainerWithContext(
+				c.id,
+				&docker.HostConfig{},
+				xcontext.IgnoreDeadline(ctx),
+			)
+			if err != nil {
+				log.Warnf(ctx, "Could not restart container %s after interrupt: %v", c.id, err)
+				return
+			}
+		case <-runDone:
+		}
+	}()
```

Comment:
I'm confused about why this works. I see the GitHub issue that we can't signal a specific `docker exec`, so a duct tape restart makes sense in the context of "we are trying to kill just the process started by `docker exec`".

But I don't follow how the restarting of the container is avoided during a Ctrl-C on `yb exec`. Or is it restarted, then immediately stopped by a different code path?

Reply:
It is the latter: it is restarted and then cleaned up later by `*Docker.Close`. The idea is to keep the Docker container running in steady-state until the caller closes the biome.

This change does not handle concurrent `Run`s well, because the stops and starts could race with each other. A better implementation would coalesce and synchronize such signals. However, this doesn't become a real issue until we try to implement ch-1251, so adding a note there.